### PR TITLE
optimize `edsClusters` `connectionNumber` to reduce access contention

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
1. use `sync.RWMutex` instead of `sync.Mutex`

1. `connectionNumber` read/write with ` sync.atomic`

### CODE REVIEW

File: pilot/pkg/proxy/envoy/v2/eds.go [hunk 1]
```diff
@@ -220,11 +221,8 @@ func localityLbEndpointsFromInstances(instances []*model.ServiceInstance) []endp
 }
 
 func connectionID(node string) string {
-	edsClusterMutex.Lock()
-	connectionNumber++
-	c := connectionNumber
-	edsClusterMutex.Unlock()
-	return node + "-" + strconv.Itoa(int(c))
+	atomic.AddInt64(&connectionNumber, 1)
+	return node + "-" + strconv.FormatInt(atomic.LoadInt64(&connectionNumber), 10)
```
Comment (ZackButcher) [inline pilot/pkg/proxy/envoy/v2/eds.go]:
```go
id := atomic.AddInt64(&connectionNumber, 1)
return node + "-" + strconv.FormatInt(id, 10)
```

Comment (hzxuzhonghu) [reply pilot/pkg/proxy/envoy/v2/eds.go]: good catch, thanks

**Comment by hzxuzhonghu** [comment]
@ZackButcher addressed, ptal

**Pull request merged**